src/Workspaces/CSharp/Portable/Simplification/Reducers/CSharpNullableAnnotationReducer.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.Simplification
+{
+    internal sealed partial class CSharpNullableAnnotationReducer : AbstractCSharpReducer
+    {
+        private static readonly ObjectPool<IReductionRewriter> s_pool = new ObjectPool<IReductionRewriter>(
+            () => new Rewriter(s_pool));
+
+        public CSharpNullableAnnotationReducer() : base(s_pool)
+        {
+        }
+
+        private static readonly Func<NullableTypeSyntax, SemanticModel, OptionSet, CancellationToken, SyntaxNode> s_simplifyNullableType = SimplifyNullableType;
+
+        private static SyntaxNode SimplifyNullableType(
+            NullableTypeSyntax node,
+            SemanticModel semanticModel,
+            OptionSet optionSet,
+            CancellationToken cancellationToken)
+        {
+            // If annotations are enabled, there's no further simplification to do
+            var context = semanticModel.GetNullableContext(node.Span.End);
```

Comment:
dumb question, why `.Span.End`?  is that what the LDM finally decided on where the `nullability` of something would be examined?

Reply:
i remember us debating on that, but i don't recall the resolution happening before i left.

Reply:
I guess I was thinking "where the question mark is" is what mattered. @333fred and @jcouv, which precise token is the one where nullability is examined to know what it is?

Reply:
Well, this was a bit of a rabbit hole :). TLDR: it looks like you have the correct code here Jason.

Example 1:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNGAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUVYAs5AsgBTEDEAOwCuAGxERgIuOQAmASygSpjZqqx4ADP2Fil0uAL0rVTAPzl4sAJTlj9O6sswAdABUA9gGUYCOQIDm3FYA3A7MPDoiIXYAvsQxQA===

In this scenario, `test` is nullable.

Example 2:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNGAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUVYAs5AsgBTEDEAOwCuAGxERgIuOQAmASygSpjZqqx4ADP2Fil0uAL0rVTeLACU5Y/WuqzMAHQAVAPYBlGAjkCA5t3MA3LbMPDoigdYAvsSRQA===

In this scenario, `test` is _oblivious_, not not nullable.

Reply:
Ah, good tests. Alright, we'll leave this as is then!